### PR TITLE
[DO NOT LAND] Switch to translation dataset in torchtext 0.9.0 release

### PR DESCRIPTION
**Pull request by zhangguanheng66**
In torchtext 0.9.0 release, we will include the raw text datasets as beta release. This PR is to switch the data to the raw dataset.

This PR should be tested against pytorch 1.8.0 rc and torchtext 0.9.0 rc.

### CODE REVIEW

**Comment by zhangguanheng66** [comment]
We expect the CI tests to fail until we switch to 1.8.0-rc.
cc @brianjo @cpuhrsch

File: beginner_source/torchtext_translation.py [hunk 1]
```diff
@@ -40,49 +40,41 @@
...
-  return Vocab(counter, specials=['<unk>', '<pad>', '<bos>', '<eos>'])
-
-de_vocab = build_vocab(train_filepaths[0], de_tokenizer)
-en_vocab = build_vocab(train_filepaths[1], en_tokenizer)
-
-def data_process(filepaths):
-  raw_de_iter = iter(io.open(filepaths[0], encoding="utf8"))
-  raw_en_iter = iter(io.open(filepaths[1], encoding="utf8"))
+
+train_iter, = Multi30k(split=('train'))
```
Comment (cpuhrsch) [inline beginner_source/torchtext_translation.py:50]: We should add support for a single string for this release as well, i.e.
```
train_iter = Multi30k(split='train')
```
I'll send a PR to torchtext.

Comment (cpuhrsch) [reply beginner_source/torchtext_translation.py:50]: We now have support for a single string argument. I'd prefer if we used that one.

**Review by cpuhrsch — CHANGES_REQUESTED** [review]
Since Multi30k is being moved into experimental for this release, let's pick a different translation dataset, for example one of the IWSLT datasets.

**Comment by zhangguanheng66** [comment]
Switch to IWSLT2017

**Review by cpuhrsch — COMMENTED** [review]
The tutorial itself still has code such as

```
# ENC_EMB_DIM = 256
# DEC_EMB_DIM = 256
# ENC_HID_DIM = 512
# DEC_HID_DIM = 512
# ATTN_DIM = 64
# ENC_DROPOUT = 0.5
# DEC_DROPOUT = 0.5
```

Did you also confirm that the model converges to something meaningful under this new dataset?

**Comment by zhangguanheng66** [comment]
> The tutorial itself still has code such as
> 
> ```
> # ENC_EMB_DIM = 256
> # DEC_EMB_DIM = 256
> # ENC_HID_DIM = 512
> # DEC_HID_DIM = 512
> # ATTN_DIM = 64
> # ENC_DROPOUT = 0.5
> # DEC_DROPOUT = 0.5
> ```
> 
> Did you also confirm that the model converges to something meaningful under this new dataset?

I'm adjusting the amount of data used in this tutorial as IWSLT2017 has more data than Multi30k

**Review by cpuhrsch — CHANGES_REQUESTED** [review]
Given the issues with convergence on this IWSLT2017 and the release of Multi30k being delayed, let's not release an update to this tutorial.

**Comment by zhangguanheng66** [comment]
Since `Multi30k` dataset is not going to be released in torchtext v0.9.0 - https://github.com/pytorch/text/pull/1207, we will delay this PR until we release `Multi30k`.

**Pull request closed without merging**